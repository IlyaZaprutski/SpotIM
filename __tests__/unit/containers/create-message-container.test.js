import React from 'react';
import configureMockStore from 'redux-mock-store';

import { MuiShallowWithContext } from '../test-utils';

import CreateMessageContainer from '../../../src/js/containers/create-message-container';

import UserReducerRecord from '../../../src/js/records/user-reducer-record';

describe('Create message container', () => {
    const initialState = {
        userInfo: new UserReducerRecord({
            userId: '0',
            userName: 'test',
            avatarUrl: 'https://spotim-demo-chat-server.herokuapp.com/avatars/001-snorlax.png',
        }),
    };
    const mockStore = configureMockStore();
    let store;
    let container;

    beforeEach(() => {
        store = mockStore(initialState);
        container = MuiShallowWithContext(<CreateMessageContainer store={store} />);
    });

    it('should render the CreateMessageForm component', () => {
        expect(container.length).toEqual(1);

        expect(container).toMatchSnapshot();
    });

    it('should render with props from initialState', () => {
        expect(container.prop('userId')).toEqual(initialState.userInfo.userId);
        expect(container.prop('userName')).toEqual(initialState.userInfo.userName);
        expect(container.prop('avatarUrl')).toEqual(initialState.userInfo.avatarUrl);
    });
});
