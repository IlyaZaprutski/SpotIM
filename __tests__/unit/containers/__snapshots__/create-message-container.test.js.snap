// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create message container should render the CreateMessageForm component 1`] = `
<CreateMessageForm
  avatarUrl="https://spotim-demo-chat-server.herokuapp.com/avatars/001-snorlax.png"
  classNames=""
  onChangeUserName={[Function]}
  onSendMessage={[Function]}
  onUpdateUserInfo={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  userId="0"
  userName="test"
/>
`;
